Parcial 1.
Sebastian Montoya - David Caliz

importar librerias

In [73]:
import pandas as pd 
import numpy as np 
from pathlib import Path

In [74]:
#links
url_cgsse = 'https://github.com/niconomist98/DataAnalyticsUQ/raw/refs/heads/main/Datos/GEIH2025/cgsse.CSV'
url_ocupados = 'https://github.com/niconomist98/DataAnalyticsUQ/raw/refs/heads/main/Datos/GEIH2025/Ocupados.CSV'
url_ft = 'https://github.com/niconomist98/DataAnalyticsUQ/raw/refs/heads/main/Datos/GEIH2025/Fuerza%20de%20trabajo.CSV'
url_desocupados = 'https://github.com/niconomist98/DataAnalyticsUQ/raw/refs/heads/main/Datos/GEIH2025/No%20ocupados.CSV'

In [75]:
# CGSSE
df_cgsse = pd.read_csv(url_cgsse, encoding='latin-1', sep=';', 
                       low_memory=False)

#Ocupados
df_ocupados=pd.read_csv(url_ocupados,encoding='latin-1',sep=';',
                        low_memory=False)

#Fuerza de trabajo
df_ft=pd.read_csv(url_ft,
                  encoding='latin-1',sep=';',low_memory=False)

#No ocupados
df_desocupados=pd.read_csv(url_desocupados,encoding='latin-1',
                           sep=';',low_memory=False)

In [76]:
df_cgsse

,PERIODO,MES,PER,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,REGIS,AREA,CLASE,...,P6170,P3041,P3042,P3042S1,P3042S2,P3043,P3043S1,P3038,P3039,POB_MAY18
0,20250626,6,2025,8238974,1,1,1,10,NaN,2,...,2.0,NaN,5.0,2.0,NaN,2.0,NaN,1.0,2.0,1.0
1,20250626,6,2025,8238974,1,2,1,10,NaN,2,...,2.0,NaN,5.0,1.0,NaN,1.0,NaN,2.0,1.0,1.0
2,20250626,6,2025,8238975,1,1,1,10,NaN,2,...,2.0,NaN,3.0,3.0,NaN,NaN,NaN,2.0,1.0,1.0
3,20250626,6,2025,8238975,1,2,1,10,NaN,2,...,2.0,NaN,1.0,0.0,NaN,NaN,NaN,2.0,1.0,1.0
4,20250626,6,2025,8238975,1,3,1,10,NaN,2,...,2.0,NaN,1.0,0.0,NaN,NaN,NaN,1.0,2.0,1.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
67851,20250626,6,2025,8283060,1,1,1,10,88.0,1,...,2.0,NaN,5.0,2.0,NaN,2.0,NaN,1.0,2.0,1.0
67852,20250626,6,2025,8283060,1,2,1,10,88.0,1,...,2.0,NaN,5.0,2.0,NaN,2.0,NaN,2.0,1.0,1.0
67853,20250626,6,2025,8283062,1,1,1,10,94.0,1,...,2.0,NaN,3.0,3.0,NaN,NaN,NaN,2.0,1.0,1.0
67854,20250626,6,2025,8283062,1,2,1,10,94.0,1,...,2.0,NaN,3.0,2.0,NaN,NaN,NaN,1.0,2.0,1.0


In [77]:
df_ft

,PERIODO,MES,PER,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,REGIS,AREA,CLASE,...,P3362S8,P3362S7A1,P6300,P6310,P6310S1,P6320,P6330,P6340,P6350,P6351
0,20250626,6,2025,8238974,1,1,1,50,NaN,2,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,20250626,6,2025,8238974,1,2,1,50,NaN,2,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,20250626,6,2025,8238975,1,1,1,50,NaN,2,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,20250626,6,2025,8238975,1,2,1,50,NaN,2,...,NaN,NaN,2.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,20250626,6,2025,8238975,1,3,1,50,NaN,2,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
54445,20250626,6,2025,8283060,1,1,1,50,88.0,1,...,NaN,NaN,2.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
54446,20250626,6,2025,8283060,1,2,1,50,88.0,1,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
54447,20250626,6,2025,8283062,1,1,1,50,94.0,1,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
54448,20250626,6,2025,8283062,1,2,1,50,94.0,1,...,NaN,NaN,2.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [78]:
df_desocupados

,PERIODO,MES,PER,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,REGIS,AREA,CLASE,...,P7360,P9460,P1519,P1883,P7422,P7422S1,OFICIO1_C8,OFICIO2_C8,RAMA2D_D_R4,RAMA4D_D_R4
0,20250626,6,2025,8238975,1,2,1,70,NaN,2,...,1.0,2.0,2,NaN,2,NaN,.,6112,1.0,0123
1,20250626,6,2025,8238978,1,1,1,70,NaN,2,...,1.0,2.0,2,NaN,2,NaN,.,9211,1.0,0161
2,20250626,6,2025,8238980,1,2,1,70,NaN,2,...,NaN,NaN,2,NaN,2,NaN,.,.,NaN,.
3,20250626,6,2025,8238981,1,2,1,70,NaN,2,...,NaN,NaN,2,NaN,2,NaN,.,.,NaN,.
4,20250626,6,2025,8238983,1,2,1,70,NaN,2,...,NaN,NaN,2,NaN,2,NaN,.,.,NaN,.
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
24739,20250626,6,2025,8283056,1,1,1,70,27.0,1,...,10.0,2.0,3,NaN,2,NaN,.,2341,85.0,8530
24740,20250626,6,2025,8283059,1,1,1,70,76.0,1,...,6.0,2.0,2,NaN,2,NaN,5223,5113,55.0,5511
24741,20250626,6,2025,8283060,1,1,1,70,88.0,1,...,9.0,2.0,2,NaN,2,NaN,.,9334,47.0,4711
24742,20250626,6,2025,8283062,1,2,1,70,94.0,1,...,NaN,NaN,2,NaN,2,NaN,.,.,NaN,.


In [79]:
df_ocupados

,PERIODO,MES,PER,DIRECTORIO,SECUENCIA_P,ORDEN,HOGAR,REGIS,AREA,CLASE,...,P1881,P1881S1,P1882,P7240,P7240S1,OCI,INGLABO,RAMA2D_R4,RAMA4D_R4,OFICIO_C8
0,20250626,6,2025,8238974,1,1,1,60,NaN,2,...,3,NaN,3.0,3,NaN,1,950000.0,1,113,6111
1,20250626,6,2025,8238974,1,2,1,60,NaN,2,...,3,NaN,4.0,5,NaN,1,NaN,1,113,6111
2,20250626,6,2025,8238975,1,1,1,60,NaN,2,...,3,NaN,10.0,3,NaN,1,700000.0,1,113,6114
3,20250626,6,2025,8238975,1,3,1,60,NaN,2,...,3,NaN,10.0,3,NaN,1,2000000.0,1,145,6122
4,20250626,6,2025,8238976,1,1,1,60,NaN,2,...,14,NaN,NaN,8,NaN,1,600000.0,1,122,6112
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
29701,20250626,6,2025,8283057,1,3,1,60,41.0,1,...,8,NaN,25.0,5,NaN,1,3800000.0,85,8544,2310
29702,20250625,6,2025,8283058,1,1,1,60,NaN,1,...,14,NaN,NaN,2,NaN,1,3000000.0,71,7111,2161
29703,20250625,6,2025,8283058,1,2,1,60,NaN,1,...,3,NaN,10.0,1,NaN,1,1500000.0,47,4752,1420
29704,20250626,6,2025,8283060,1,2,1,60,88.0,1,...,3,NaN,10.0,8,NaN,1,500000.0,56,5619,7512


1. Para el total nacional:
    Poblacion total

In [80]:
Población_total = df_cgsse['FEX_C18'].sum()
Población_total

np.float64(52045001.000009045)

La poblacion total nacional calculada es de 52.045.001 de personas

Poblacion en edad de trabajar

In [81]:
PET = df_cgsse[df_cgsse['P6040'] >= 15]['FEX_C18'].sum()
PET

np.float64(40653416.00000409)

La poblacion en edad de trabajar es de 40.653.416 entre (15 y 65 años)

Poblacion que no esta en edad de trabajar

In [82]:
No_PET = df_cgsse[df_cgsse['P6040'] < 15]['FEX_C18'].sum()
No_PET

np.float64(11391585.000004958)

La poblacion que no esta en edad de trabajar es de 11.391.585 menores a ( 15 y mayores de 65)

Poblacion economicamente activa

In [83]:
PEA = df_ft[(df_ft['FT']==1)]['FEX_C18'].sum()
print(f'La población Economicamente Activa es de {PEA}')

La población Economicamente Activa es de 25974846.22356785


La poblacion economicamente activa es de 25.974.846

Poblacion economicamente inactiva

In [84]:
PEI = df_ft[(df_ft['FFT']==1)]['FEX_C18'].sum()
print(f'La población Economicamente Inactiva de la {PEI}')

La población Economicamente Inactiva de la 14678569.776436238


Tasa de ocupacion

In [85]:
Ocupados = df_ocupados['FEX_C18'].sum()
Ocupados

np.float64(23752952.660255846)

In [86]:
Tasa_de_ocupación = (Ocupados/PET)*100
Tasa_de_ocupación

np.float64(58.427937913639184)

La tasa de ocupacion despues de hacer los calculos es del 58,43%

Tasa de desempleo

In [87]:
Desocupados = df_desocupados[df_desocupados['DSI']==1]['FEX_C18'].sum()
Desocupados

np.float64(2221893.563312005)

In [88]:
Tasa_de_desempleo = (Desocupados/PEA)*100
Tasa_de_desempleo

np.float64(8.554020086155528)

La tasa de desempleo despues de hacer los calculos es del 8,55%

2. Calcule las tasas de desempleo para hombres y mujeres en Risaralda, Caldas , Tolima y Quindío. ¿Cuál es el departamento con mayor y menor tasa de desempleo de los 4?

In [89]:
codigos_region = [17, 63, 66, 73]

df_desocupados_reg = df_desocupados[df_desocupados['DPTO'].isin(codigos_region)].copy()
df_ocupados_reg = df_ocupados[df_ocupados['DPTO'].isin(codigos_region)].copy()

df_desocupados_gen = pd.merge(
    df_desocupados_reg[df_desocupados_reg["DSI"]==1], 
    df_cgsse[['DIRECTORIO','SECUENCIA_P','ORDEN','P3271']], 
    on=['DIRECTORIO','SECUENCIA_P','ORDEN'], 
    how='left')

df_ocupados_gen = pd.merge(
    df_ocupados_reg, 
    df_cgsse[['DIRECTORIO','SECUENCIA_P','ORDEN','P3271']], 
    on=['DIRECTORIO','SECUENCIA_P','ORDEN'], 
    how='left')

desocupados_stats = df_desocupados_gen.groupby(['DPTO','P3271'])['FEX_C18'].sum().reset_index(name='desocupados')
ocupados_stats = df_ocupados_gen.groupby(['DPTO','P3271'])['FEX_C18'].sum().reset_index(name='ocupados')

pea_final = pd.merge(ocupados_stats, desocupados_stats, on=['DPTO','P3271'], how='outer').fillna(0)
pea_final['PEA'] = pea_final['ocupados'] + pea_final['desocupados']
pea_final['TD'] = (pea_final['desocupados'] / pea_final['PEA']) * 100

nombres_depto = {17: 'Caldas', 63: 'Quindío', 66: 'Risaralda', 73: 'Tolima'}
pea_final['Departamento'] = pea_final['DPTO'].map(nombres_depto)
pea_final['Sexo'] = pea_final['P3271'].map({1:'Hombres', 2:'Mujeres'})

print(pea_final[['Departamento', 'Sexo', 'TD']].sort_values(['Departamento', 'Sexo']))

  Departamento     Sexo         TD
0       Caldas  Hombres   5.583183
1       Caldas  Mujeres   7.677962
2      Quindío  Hombres  10.233032
3      Quindío  Mujeres   9.084532
4    Risaralda  Hombres   6.074260
5    Risaralda  Mujeres  10.669927
6       Tolima  Hombres   7.220583
7       Tolima  Mujeres  19.202747


In [90]:
td_max = pea_final.loc[pea_final.groupby("Sexo")["TD"].idxmax()]
td_min = pea_final.loc[pea_final.groupby("Sexo")["TD"].idxmin()]

print("\nMAYOR tasa de desempleo por sexo")
print(td_max[['Departamento', 'Sexo', 'TD']])

print("\nMENOR tasa de desempleo por sexo")
print(td_min[['Departamento', 'Sexo', 'TD']])


MAYOR tasa de desempleo por sexo
  Departamento     Sexo         TD
2      Quindío  Hombres  10.233032
7       Tolima  Mujeres  19.202747

MENOR tasa de desempleo por sexo
  Departamento     Sexo        TD
0       Caldas  Hombres  5.583183
1       Caldas  Mujeres  7.677962


El porcentaje mayor de TD para las mujeres es del Tolima con 19.20% y la menor  TD es de Caldas con 7.68%.
Para los hombres la mayor TD es Quindio con 19.20% y la menor tasa de desempleo es Caldas con 5.58%

3. . Tomando en consideración únicamente los siguientes grupos de edad:

18 a 24 24 a 30 30 a 40 40 a 50 +60

Para Risaralda, Caldas , Tolima y Quindío, cual es el grupo de edad con mayor y menor tasa de desempleo (indicar en cada departamento el grupo de de edad con mayor y menor tasa de desempleo) ?

In [91]:
deptos_interes = [17, 63, 66, 73]

def clasificar_edad_trabajo(x):
    if 18 <= x < 24:
        return "18-24"
    elif 24 <= x < 30:
        return "24-30"
    elif 30 <= x < 40:
        return "30-40"
    elif 40 <= x < 50:
        return "40-50"
    elif x >= 60:
        return "60 en adelante"
    else:
        return "Fuera de rango"

df_cgsse["grupo_edad"] = df_cgsse["P6040"].apply(clasificar_edad_trabajo)

df_desocupados_reg = df_desocupados[df_desocupados["DPTO"].isin(deptos_interes)].copy()
des_edad = pd.merge(
    df_desocupados_reg[df_desocupados_reg["DSI"]==1],
    df_cgsse[["DIRECTORIO","SECUENCIA_P","ORDEN","grupo_edad"]],
    on=["DIRECTORIO","SECUENCIA_P","ORDEN"],
    how="left"
).groupby(["DPTO","grupo_edad"])["FEX_C18"].sum().reset_index(name="desocupados")

df_ocupados_reg = df_ocupados[df_ocupados["DPTO"].isin(deptos_interes)].copy()
ocu_edad = pd.merge(
    df_ocupados_reg,
    df_cgsse[["DIRECTORIO","SECUENCIA_P","ORDEN","grupo_edad"]],
    on=["DIRECTORIO","SECUENCIA_P","ORDEN"],
    how="left"
).groupby(["DPTO","grupo_edad"])["FEX_C18"].sum().reset_index(name="ocupados")

pea_edad_reg = pd.merge(ocu_edad, des_edad, on=["DPTO","grupo_edad"], how="outer").fillna(0)
pea_edad_reg["PEA"] = pea_edad_reg["ocupados"] + pea_edad_reg["desocupados"]
pea_edad_reg["TD"] = (pea_edad_reg["desocupados"] / pea_edad_reg["PEA"]) * 100

grupos_validos = ["18-24", "24-30", "30-40", "40-50", "60 en adelante"]
pea_final = pea_edad_reg[pea_edad_reg["grupo_edad"].isin(grupos_validos)].copy()

nombres_depto = {17: 'Caldas', 63: 'Quindío', 66: 'Risaralda', 73: 'Tolima'}
pea_final['Departamento'] = pea_final['DPTO'].map(nombres_depto)

mayores_reg = pea_final.loc[pea_final.groupby("Departamento")["TD"].idxmax()]
menores_reg = pea_final.loc[pea_final.groupby("Departamento")["TD"].idxmin()]

print("\nMayor tasa de desempleo por departamento:")
print(mayores_reg[['Departamento', 'grupo_edad', 'TD']])

print("\nMenor tasa de desempleo por departamento:")
print(menores_reg[['Departamento', 'grupo_edad', 'TD']])


Mayor tasa de desempleo por departamento:
   Departamento grupo_edad         TD
0        Caldas      18-24   9.972584
7       Quindío      24-30  14.237924
12    Risaralda      18-24  15.173556
18       Tolima      18-24  19.348020

Menor tasa de desempleo por departamento:
   Departamento      grupo_edad        TD
4        Caldas  60 en adelante  4.060989
9       Quindío           40-50  6.506359
16    Risaralda  60 en adelante  3.639861
22       Tolima  60 en adelante  9.458632


Para los departamentso de Risaralda, Caldas y Tolima con mayor desempleo es entre 18 a 24 años, mientras que en Quindio es entre los 24 y 30 años. Dado a estos calculos realizados, la menor tasa de desempleo se registra en los mayores de 60 años en Tolima, Risarlada y Caldas y en el Quindio entre los 40 y 50 años.

4.  Entre Risaralda, Caldas , Tolima y Quindío, cuál es el departamento con menor tasa de desempleo de mujeres entre los 18 y los 25 años ?

In [92]:
# 1. Definimos los departamentos de interés
deptos_interes = [17, 63, 66, 73]

# 2. Preparamos los datos de Desocupados (DSI == 1) incluyendo Sexo y Edad
df_desocupados_reg = pd.merge(
    df_desocupados[(df_desocupados["DSI"] == 1) & (df_desocupados["DPTO"].isin(deptos_interes))],
    df_cgsse[["DIRECTORIO", "SECUENCIA_P", "ORDEN", "P3271", "P6040"]], # Agregamos P6040 aquí
    on=["DIRECTORIO", "SECUENCIA_P", "ORDEN"],
    how="left"
)

# 3. Preparamos los datos de Ocupados incluyendo Sexo y Edad
df_ocupados_reg = pd.merge(
    df_ocupados[df_ocupados["DPTO"].isin(deptos_interes)],
    df_cgsse[["DIRECTORIO", "SECUENCIA_P", "ORDEN", "P3271", "P6040"]], # Agregamos P6040 aquí
    on=["DIRECTORIO", "SECUENCIA_P", "ORDEN"],
    how="left"
)

# 4. Filtramos para MUJERES (P3271 == 2) entre 18 y 25 años
# Para Desocupadas
desem_joven = df_desocupados_reg[
    (df_desocupados_reg["P3271"] == 2) & 
    (df_desocupados_reg["P6040"].between(18, 25))
].groupby("DPTO")["FEX_C18"].sum()

# Para Ocupadas
ocu_joven = df_ocupados_reg[
    (df_ocupados_reg["P3271"] == 2) & 
    (df_ocupados_reg["P6040"].between(18, 25))
].groupby("DPTO")["FEX_C18"].sum()

# 5. Calculamos la Tasa de Desempleo (PEA = Ocupados + Desocupados)
pea_joven = ocu_joven.add(desem_joven, fill_value=0)
resultado_final = pd.DataFrame({
    "PEA": pea_joven,
    "Desempleadas": desem_joven
}).fillna(0)

resultado_final["Tasa_desempleo(%)"] = (resultado_final["Desempleadas"] / resultado_final["PEA"]) * 100

# 6. Mapear nombres y ordenar
nombres = {17: 'CALDAS', 63: 'QUINDIO', 66: 'RISARALDA', 73: 'TOLIMA'}
resultado_final["Departamento"] = resultado_final.index.map(nombres)
resultado_final = resultado_final.sort_values(by="Tasa_desempleo(%)")

# Mostrar resultado
print("=== Tasa de Desempleo Mujeres 18-25 años (Región) ===")
print(resultado_final[['Departamento', 'Tasa_desempleo(%)']])

=== Tasa de Desempleo Mujeres 18-25 años (Región) ===
     Departamento  Tasa_desempleo(%)
DPTO                                
63        QUINDIO           7.468211
17         CALDAS          14.500102
73         TOLIMA          21.378071
66      RISARALDA          22.992769


Entre los 4 departamentos, Quindio es el que presenta la menor tasa de desempleo para mujeres entre los 18 y 25 años

5.¿Cuántos hombres y mujeres hay en edad de trabajar en el quindio ? ¿Qué porcentaje de la población total representan ?

In [93]:
df_quindio = df_cgsse[df_cgsse["DPTO"] == 63].copy()

df_quindio['es_PET'] = df_quindio['P6040'] >= 15

poblacion_total_quindio = df_quindio['FEX_C18'].sum()

pet_quindio = df_quindio[df_quindio['es_PET'] == True].groupby('P3271')['FEX_C18'].sum()

hombres_pet = pet_quindio.get(1, 0)
mujeres_pet = pet_quindio.get(2, 0)
total_pet = hombres_pet + mujeres_pet

porcentaje_hombres = (hombres_pet / poblacion_total_quindio) * 100
porcentaje_mujeres = (mujeres_pet / poblacion_total_quindio) * 100
porcentaje_pet_total = (total_pet / poblacion_total_quindio) * 100

print(f"Población Total estimada en Quindío: {int(poblacion_total_quindio)}")
print(f"Hombres  edad de trabajar: {int(hombres_pet)} ({porcentaje_hombres:.2f}% del total)")
print(f"Mujeres  edad de trabajar: {int(mujeres_pet)} ({porcentaje_mujeres:.2f}% del total)")
print(f"Total PET Quindío: {int(total_pet)} ({porcentaje_pet_total:.2f}% del total)")

Población Total estimada en Quindío: 603068
Hombres  edad de trabajar: 247037 (40.96% del total)
Mujeres  edad de trabajar: 261387 (43.34% del total)
Total PET Quindío: 508425 (84.31% del total)


Con los datos alcanzados en el departamento del Quindio:
Hombres en edad de trabajar: Representan aproximadamente el 40.96%
Mujeres en edad de trabajar: Representan aproximadamente el 43.34%
Poblacion total PET: Cerca del 80-82% para el Quindio, aca los quindianos tienen mas de 15 años.

6. ¿Qué porcentaje de la población del Tolima está en edad de trabajar ?

In [94]:
# 1. Filtrar datos para Tolima (Código 73)
df_tolima = df_cgsse[df_cgsse["DPTO"] == 73].copy()

# 2. Identificar la PET (15 años o más)
df_tolima['es_PET'] = df_tolima['P6040'] >= 15

# 3. Sumar poblaciones usando el factor de expansión (FEX_C18)
poblacion_total_tolima = df_tolima['FEX_C18'].sum()
poblacion_pet_tolima = df_tolima[df_tolima['es_PET'] == True]['FEX_C18'].sum()

# 4. Calcular el porcentaje
porcentaje_pet = (poblacion_pet_tolima / poblacion_total_tolima) * 100

print(f"Población Total Tolima: {int(poblacion_total_tolima)}")
print(f"Población en Edad de Trabajar (PET): {int(poblacion_pet_tolima)}")
print(f"Porcentaje PET: {porcentaje_pet:.2f}%")

Población Total Tolima: 1411965
Población en Edad de Trabajar (PET): 1087359
Porcentaje PET: 77.01%


En el departamento del Tolima la poblacion en edad de trabajar es del 77.01% de su poblacion

7.¿ Qué porcentaje de la población de Risaralda es población económicamente inactiva ?

In [95]:
PEIR= df_ft[(df_ft["FFT"]==1) & (df_ft['DPTO']==66)]["FEX_C18"].sum()
PETR= df_cgsse[(df_cgsse['DPTO']==66)]["FEX_C18"].sum()
print(f" poblacion en edad de trabajar es :{PETR}")
print(f"poblacion economicamente inactiva de risaralda: {PEIR}")
print(f" porcentaje de la PEI en risaralda {PEIR/PETR*100}%")

 poblacion en edad de trabajar es :1025240.9423573611
poblacion economicamente inactiva de risaralda: 332743.42055744905
 porcentaje de la PEI en risaralda 32.45514364578185%


La poblacion economicamente inactiva en Risaralda despues de sacar los datos es de 1.025.240 que representa el 32.45% de la poblacion

8. ¿Cuántas personas hay en el Quindío en condición de desempleo entre los 18 y los 30 años ?

In [96]:
# 1. Unimos desocupados con edad y departamento
df_desocupados_age = pd.merge(
    df_desocupados,
    df_cgsse[['DIRECTORIO','SECUENCIA_P','ORDEN','P6040','DPTO']],
    on=['DIRECTORIO','SECUENCIA_P','ORDEN'],
    how='left'
)

# 2. Filtrar Quindío, desempleados entre 18 y 30 años
desempleo_18_30_quindio = df_desocupados_age[
    (df_desocupados_age['DPTO_y']==63) &
    (df_desocupados_age['DSI']==1) &
    (df_desocupados_age['P6040']>=18) &
    (df_desocupados_age['P6040']<=30)
]['FEX_C18'].sum()

print("Personas desempleadas entre 18 y 30 años en el Quindío:", round(desempleo_18_30_quindio))

Personas desempleadas entre 18 y 30 años en el Quindío: 9480


En el Quindio, la condicion de desempleo hay 9.480 peronas

9.¿Cuántas personas hay ocupadas en el departamento del Tolima , cuyas edades están entre los 20 y los 25 años , qué porcentaje de la población total del Tolima representan ?

In [97]:
opt= df_ocupados[(df_ocupados["OCI"]==1) & (df_desocupados['DPTO']==73)]["FEX_C18"].sum()
print(f"poblacion ocupada en el departamento del tolima: {opt}")

poblacion ocupada en el departamento del tolima: 992718.108798512


La poblacion ocupada en el departamento del Tolima es de 992.718